browser/brave_rewards/rewards_extension_panel_handler.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_rewards/rewards_extension_panel_handler.h"
+
+#include <memory>
+#include <string>
+
+#include "base/strings/strcat.h"
+#include "brave/browser/brave_rewards/rewards_service_factory.h"
+#include "brave/browser/extensions/brave_component_loader.h"
+#include "brave/browser/ui/views/brave_actions/brave_actions_container.h"
+#include "chrome/browser/extensions/extension_service.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/browser_window.h"
+#include "extensions/browser/extension_system.h"
+
+namespace brave_rewards {
+
+namespace {
+
+constexpr char kRewardsPanelUrl[] = "/brave_rewards_panel.html";
+constexpr char kAdsEnableRelativeUrl[] = "/request_ads_enabled_panel.html";
+
+std::string GetExtensionPath(const mojom::RewardsPanelArgs& args) {
+  switch (args.view) {
+    case mojom::RewardsPanelView::kDefault:
+      return kRewardsPanelUrl;
+    case mojom::RewardsPanelView::kRewardsTour:
+      return base::StrCat({kRewardsPanelUrl, "#tour"});
+    case mojom::RewardsPanelView::kGrantCaptcha:
+      return base::StrCat({kRewardsPanelUrl, "#grant_", args.data});
+    case mojom::RewardsPanelView::kAdaptiveCaptcha:
+      return base::StrCat({kRewardsPanelUrl, "#load_adaptive_captcha"});
+    case mojom::RewardsPanelView::kBraveTalkOptIn:
+      return kAdsEnableRelativeUrl;
+  }
+}
+
+}  // namespace
+
+RewardsExtensionPanelHandler::~RewardsExtensionPanelHandler() = default;
+
+bool RewardsExtensionPanelHandler::IsRewardsExtensionPanelURL(const GURL& url) {
+  return url.SchemeIs("chrome-extension") &&
+         url.host() == brave_rewards_extension_id &&
+         (url.path() == kRewardsPanelUrl ||
+          url.path() == kAdsEnableRelativeUrl);
+}
+
+void RewardsExtensionPanelHandler::OnRewardsPanelRequested(
+    Browser* browser,
+    const mojom::RewardsPanelArgs& args) {
+  DCHECK(browser);
+  auto* profile = browser->profile();
+
+  // Start the rewards ledger process if it is not already started
+  auto* rewards_service = RewardsServiceFactory::GetForProfile(profile);
+  if (!rewards_service) {
+    return;
+  }
+
+  rewards_service->StartProcess(base::DoNothing());
```

Comment:
doesn't this run at startup/enable? Why would it ever not be running at this point?

Reply:
The Rewards utility process is only started at startup if Ads or AC is enabled. This is also existing behavior for the extension  panel so I didn't want to disrupt that in this PR.